modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/dsl/metrics/StatsDReporter.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2021-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.dsl.metrics
+
+import java.net.{DatagramPacket, DatagramSocket, InetAddress}
+import java.nio.charset.StandardCharsets.UTF_8
+
+import cats.implicits._
+
+import cats.effect.{Blocker, ContextShift, Resource, Sync, Timer}
+
+import com.snowplowanalytics.snowplow.rdbloader.common.config.Config
+
+object StatsDReporter {
+
+  def build[F[_]: ContextShift: Sync: Timer](statsDConfig: Option[Config.StatsD], blocker: Blocker): Reporter[F] = 
+    statsDConfig match {
+      case Some(config) =>
+        new Reporter[F] {
+          def report(metrics: List[Metrics.KVMetric]): F[Unit] = {
+            mkSocket[F](blocker).use { socket =>
+              for {
+                formatted <- Sync[F].pure(metrics.map(statsDFormat(config)))
+                ip <- blocker.delay(InetAddress.getByName(config.hostname))
+                _ <- formatted.traverse_(sendMetric[F](blocker, socket, ip, config.port))
+              } yield ()
```

Comment:
```suggestion
          def report(metrics: List[Metrics.KVMetric]): F[Unit] = {
            val formatted = metrics.map(statsDFormat(config))
            mkSocket[F](blocker).use { socket =>
              for {
                ip <- blocker.delay(InetAddress.getByName(config.hostname))
                _ <- formatted.traverse_(sendMetric[F](blocker, socket, ip, config.port))
              } yield ()
```

This is obviously super minor in this case, but generally you'd want to keep as few things inside `IO` as possible, especially if `ContextShift` is used.